<launch>
  
  <!-- laod multi detection -->
  <node pkg="tuw_artoolkitplus" type="artoolkitplus_node" name="arMarker" output="screen">

    <param name="skip_frames" value="0"/> <!-- to skip every x frames 0 means no skip -->
    <param name="show_camera_image" value="true"/> <!-- shows a debug image, it slows the system down -->
    <!--param name="tf_prefix" value="artags"/ --> <!-- tf prefix, if not set it will use the node name as prefix  -->
    
    <param name="tracker_single_marker" value="false"/>  <!-- detect single patterns  -->
    
    <param name="tracker_multi_marker" value="true"/>  <!-- using a multi pattern detection  -->
    <param name="pattern_frame" value="pattern"/> <!-- frame name of the multi pattern  -->
    <param name="pattern_file" value="$(find tuw_artoolkitplus)/cfg/markerboard_0000-0011.cfg"/> <!-- pattern file used  -->
    
    
    <param name="marker_mode" value="bch"/> <!-- type of marker encoding bch or simple are avaliable -->
    <param name="pattern_width" value="0.1"/> <!-- marker size in meter -->
    <param name="threshold" value="0"/>  <!-- threshold for marker detection, 0 means auto  -->
    <param name="border_width" value="0"/> <!-- inner border to the encoding, 0 means auto  -->
    <param name="undist_iterations" value="10"/>  <!-- number of interation to undistore a corner  -->
    <param name="distorted_input" value="true"/>  <!-- true if you are working with distorted raw images, false if the image is undistorted  -->
    <param name="undist_mode" value="std"/>  <!-- undistor modes: none, std or lut (lut only works up to 1024x1024 -->
    <param name="pose_estimation_mode" value="rpp"/>  <!-- pose estimation modes: normal, cont or rpp -->

    <remap from="image" to="camera/image_raw"/>
    <remap from="camera_info" to="camera/camera_info"/>    

  </node>

</launch>

